src/libs/Navigation/AppNavigator/MainDrawerNavigator.js

```diff
@@ -111,6 +114,13 @@ class MainDrawerNavigator extends Component {
         if (!this.initialParams.reportID) {
             const state = this.props.navigation.getState();
             const reportScreenKey = lodashGet(state, 'routes[0].state.routes[0].key', '');
+
+            // If this onyx key exists then it means we stashed the url from a deep link
+            // so we want to override the reportID that is being loaded by default to the one the user was attempting to load initially.
+            const reportScreenReportID = this.props.reportScreenReportID;
+            if (!_.isEmpty(reportScreenReportID)) {
+                initialNextParams.reportID = reportScreenReportID;
```

Comment:
@bondydaa I suggest we will store route from url and if it not empty we will navigate to it in here after login success. Here is result

https://user-images.githubusercontent.com/129500732/236386087-25af0c50-7149-42b3-8a6c-a15398a218d7.mp4